### PR TITLE
Enable GitHub auth on Netlify CMS

### PR DESCRIPTION
**Pull request by esirK**
## Description

Adds Github Authentication on our Netlify CMS

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Screenshots
Screenshot 2022-07-27 at 11 38 11

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/pages/api/admin/config.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 module.exports = {
   backend: {
-    name: "git-gateway",
+    name: "github",
     branch: "main",
+    repo: "CodeForAfrica/ui",
```
Comment (kilemensi) [inline apps/codeforafrica/src/pages/api/admin/config.js:5]: Can't we set these up in the [`config.yml.page`](https://github.com/CodeForAfrica/ui/blob/b96a2604e2f8393f5246522c69580272e12bc1fb/apps/codeforafrica/src/pages/api/admin/config.yml.page.js#L9-L19) for production use only i.e. default should still be local development

File: apps/codeforafrica/src/pages/api/oauth/auth.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      )}/api/oauth/callback`,
+      scope: "user, repo",
+    },
+  },
+};
+
+const client = new AuthorizationCode({
+  client: {
+    id: process.env.OAUTH_CLIENT_ID,
+    secret: process.env.OAUTH_CLIENT_SECRET,
```
Comment (kilemensi) [inline apps/codeforafrica/src/pages/api/oauth/auth.js:21]: These should be added in the `env.template` for documentation.

**Review by kilemensi — APPROVED** [review]
🚀

File: apps/codeforafrica/src/pages/api/admin/config.js (hunk 1 above)
Comment (esirK) [reply apps/codeforafrica/src/pages/api/admin/config.js:5]: With this setup, the default is still `local` just that we don't have to use the `git-gateway` backend

Comment (kilemensi) [reply apps/codeforafrica/src/pages/api/admin/config.js:5]: Cool, cool... I think the setup was copied from the [docs](https://www.netlifycms.org/docs/beta-features/#working-with-a-local-git-repository) directly. e.g. not sure if we need `proxy_url` either.

**Pull request merged**